terraform/gcp/modules/argocd/argocd.tf

```diff
@@ -123,38 +124,54 @@ resource "helm_release" "argocd_apps" {
   ]
 }
 
+# this needs to be manually granted access via admin console
+# - Client ID needs to be enabled for domain-wide delegation
+# - SA needs to be granted "Groups Reader" role
 resource "google_service_account" "argocd-directory-api-sa" {
   account_id   = "argocd-directory-api-sa"
   display_name = "ArgoCD Directory API Service Account"
   project      = var.project_id
 }
 
-resource "kubectl_manifest" "externalsecret_argocd_directory_api_credentials" {
+resource "kubectl_manifest" "externalsecret_argocd_oauth_client_credentials" {
   yaml_body = <<YAML
 apiVersion: external-secrets.io/v1beta1
 kind: ExternalSecret
 metadata:
-  name: gcp-external-secret-argocd-directory-api-credentials
+  name: gcp-external-secret-argocd-oauth-client-credentials
   namespace: "${kubernetes_namespace_v1.argocd.metadata[0].name}"
 spec:
   secretStoreRef:
     kind: ClusterSecretStore
     name: gcp-backend
   target:
-    name: argocd-directory-api-credentials
+    name: argocd-oauth-client-credentials
   data:
-  - secretKey: googleAuth.json
+  - secretKey: client-id
     remoteRef:
-      key: "${var.gcp_secret_name_directory_api_credentials}"
+      key: "${google_secret_manager_secret.argocd-oauth-client-id.secret_id}"
```

Comment:
I assume this will get manually uploaded before applying terraform?

Reply:
after, since terraform is creating the secret object itself.